source/_posts/2015-03-23-artsy-technology-stack-2015.markdown

```diff
@@ -0,0 +1,52 @@
+blog-url: http://code.dblock.org
+---
+Artsy has now grown past 100 team members and our Engineering organization is now 20 strong. For a brief overview of what the company has accomplished in the last two years, check out our [2013](http://2013.artsy.net) and [2014](http://2014.artsy.net) reviews.
+
+This is a good opportunity to describe our updated technology stack. Last time [we did this](/blog/2012/10/10/artsy-technology-stack) was when Artsy launched publicly in 2012.
+
+<center><img src='/images/2015-03-23-artsy-technology-stack-2015/stats.png'></center>
+
+Three years ago Artsy was a classic Ruby-on-Rails monolith with a handful of adjacent processes and tools. We've since broken it up into many independent services, and continue to heavily be a Ruby and JavaScript shop, using Rails where appropriate, with native code on mobile devices and some JVM-based experiments in micro-services.
+
+<center><img src='/images/2015-03-23-artsy-technology-stack-2015/languages.png'></center>
+
+<!-- more -->
+
+What you see today when you go to [www.artsy.net](https://www.artsy.net) is a website built with [Ezel.js](http://ezeljs.com), which is a boilerplate for Backbone projects running on Node.js and using Express and Browserify. The [CoffeeScript code](https://github.com/artsy/force-public) is open-source. The mobile version of www.artsy.net is [m.artsy.net](https://m.artsy.net) and is built on the same technology. Both run on Heroku and use [Redis](http://redis.io) for caching.
+
+<center><img src='/images/2015-03-23-artsy-technology-stack-2015/artsy.png'></center>
+
+These web applications talk to a private Ruby API built with [Grape](https://github.com/intridea/grape), that serves JSON. We also have a more modern and better designed [public HAL+JSON API](https://developers.artsy.net). For historical reasons, both are hosted side-by-side on top of a big Rails app that used to be our original monolith. The API service runs on [AWS OpsWorks](http://aws.amazon.com/opsworks) and retrieves data from several [MongoDB](http://www.mongodb.com) databases hosted with [Compose](https://www.compose.io). It also uses [Apache Solr](http://lucene.apache.org/solr), being replaced by [Elastic Search](https://www.elastic.co) and [Google Custom Search](https://www.google.com/cse). The API service also heavily relies on [Memcached](http://memcached.org).
+
+<center><img src='/images/2015-03-23-artsy-technology-stack-2015/developers.png'></center>
+
+Artwork, artist and other metadata is uploaded by our partners via a home-grown content-management system built entirely from scratch in 2014 on Ruby-on-Rails. This was a deliberate "boring" choice that continues to serve us well. We have adopted a common model for admin-type apps with a shared component library and a reusable UI, all implemented as Rails engines. Using these components we are able to quickly and easily compose applications to manage fairs or auctions, for example.
```

Comment:
Gotta take that plural home through... "Artwork**s**, artist**s** and other metadata **are** uploaded by our partners..."
Why not make it active voice? Oxford comma? (kidding)
"Our partners upload artworks, artists and other metadata via a home-grown content-management system built..."

Also - I'd like a PE blog post from @dylanfareed :)